sockeye/training.py

```diff
@@ -304,20 +310,31 @@ def _fit(self,
                         os.remove(best_path)
                     actual_best_fname = C.PARAMS_NAME % best_checkpoint
                     os.symlink(actual_best_fname, best_path)
-                    training_state.num_not_improved = 0
+                    train_state.num_not_improved = 0
                 else:
-                    training_state.num_not_improved += 1
-                    logger.info("Model has not improved for %d checkpoints", training_state.num_not_improved)
+                    train_state.num_not_improved += 1
+                    logger.info("Model has not improved for %d checkpoints", train_state.num_not_improved)
+
+                stop_fit = False
+
+                if train_state.num_not_improved == max_num_not_improved:
+                    logger.info("Stopping fit (no improvements for %d checkpoints)", train_state.num_not_improved)
+                    stop_fit = True
+
+                if min_num_epochs is not None and train_state.epoch < min_num_epochs:
+                    logger.info("Minimum number of epochs (%d) not reached: %d. Will not stop yet.",
+                                min_num_epochs,
+                                train_state.epoch)
+                    stop_fit = False
```

Comment:
Couldn't we end up with contradicting messages here? "Stopping fit (no improvement)" followed with "Sorry, won't stop yet!". I would prefer a single message. And also only show the "Minimum number of epochs" message if we would stop had the option be deactivated.